engine/access/rpc/backend/backend_stream_transactions_test.go

```diff
@@ -0,0 +1,343 @@
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine"
+	access "github.com/onflow/flow-go/engine/access/mock"
+	backendmock "github.com/onflow/flow-go/engine/access/rpc/backend/mock"
+	connectionmock "github.com/onflow/flow-go/engine/access/rpc/connection/mock"
+	"github.com/onflow/flow-go/engine/access/subscription"
+	subscriptionmock "github.com/onflow/flow-go/engine/access/subscription/mock"
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/counters"
+	"github.com/onflow/flow-go/module/metrics"
+	protocol "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type TransactionStatusSuite struct {
+	suite.Suite
+
+	state          *protocol.State
+	sealedSnapshot *protocol.Snapshot
+	finalSnapshot  *protocol.Snapshot
+	log            zerolog.Logger
+
+	blocks             *storagemock.Blocks
+	headers            *storagemock.Headers
+	collections        *storagemock.Collections
+	transactions       *storagemock.Transactions
+	receipts           *storagemock.ExecutionReceipts
+	results            *storagemock.ExecutionResults
+	transactionResults *storagemock.LightTransactionResults
+	seals              *storagemock.Seals
+
+	colClient              *access.AccessAPIClient
+	execClient             *access.ExecutionAPIClient
+	historicalAccessClient *access.AccessAPIClient
+	archiveClient          *access.AccessAPIClient
+
+	connectionFactory *connectionmock.ConnectionFactory
+	communicator      *backendmock.Communicator
+	blockTracker      *subscriptionmock.BlockTracker
+	reporter          *syncmock.IndexReporter
+
+	chainID flow.ChainID
+
+	broadcaster    *engine.Broadcaster
+	rootBlock      flow.Block
+	sealedBlock    *flow.Block
+	finalizedBlock *flow.Block
+
+	blocksArray []*flow.Block
+	blockMap    map[uint64]*flow.Block
+	resultsMap  map[flow.Identifier]*flow.ExecutionResult
+
+	backend *Backend
+}
+
+func TestTransactionStatusSuite(t *testing.T) {
+	suite.Run(t, new(TransactionStatusSuite))
+}
+
+// SetupTest initializes the test suite with required dependencies.
+func (s *TransactionStatusSuite) SetupTest() {
+	s.log = zerolog.New(zerolog.NewConsoleWriter())
+	s.state = new(protocol.State)
+	s.sealedSnapshot = new(protocol.Snapshot)
+	s.finalSnapshot = new(protocol.Snapshot)
+	header := unittest.BlockHeaderFixture()
+
+	params := new(protocol.Params)
+	params.On("FinalizedRoot").Return(header, nil)
+	params.On("SporkID").Return(unittest.IdentifierFixture(), nil)
+	params.On("ProtocolVersion").Return(uint(unittest.Uint64InRange(10, 30)), nil)
+	params.On("SporkRootBlockHeight").Return(header.Height, nil)
+	params.On("SealedRoot").Return(header, nil)
+	s.state.On("Params").Return(params)
+
+	s.blocks = new(storagemock.Blocks)
```

Comment:
Is this copied from somewhere? I am a strong advocate of using `storagemock.NewT` instead of `new(T)` because the first one enforces assertions. If this is copied then did you try to extract a common part so we don't make an identical copy so later one when we fix those tests we do it in one place. If this is a brand new test suite which doesn't depend on any functionality then I am suggesting to not introducing tech dept and implement a safer API for tests.